verticapy/learn/vmodel.py

```diff
@@ -2548,26 +2575,20 @@ def fit(
         if self.type in (
             "RandomForestClassifier",
             "RandomForestRegressor",
-            "XGBoostRegressor",
+            "XGBoostClassifier",
             "XGBoostRegressor",
         ) and isinstance(verticapy.options["random_state"], int):
             query += ", seed={}, id_column='{}'".format(
-                verticapy.options["random_state"], X[0]
-            )
-        if self.type == "BisectingKMeans" and isinstance(
-            verticapy.options["random_state"], int
-        ):
-            query += ", kmeans_seed={}, id_column='{}'".format(
-                verticapy.options["random_state"], X[0]
+                verticapy.options["random_state"], id_column_name
             )
         query += ")"
         try:
             executeSQL(query, title="Fitting the model.")
-            if isinstance(input_relation, vDataFrame):
-                drop_if_exists(relation, method="view")
+            if tmp_view:
+                drop(relation, method="view")
```

Comment:
Isn't better to have `drop_if_exists`?

Reply:
We are sure that it exists as it was created previously with an unique id